indispensable-cosef/src/commonMain/kotlin/at/asitplus/signum/indispensable/cosef/CoseSignedBytes.kt

```diff
@@ -32,11 +32,12 @@ data class CoseSignedBytes(
 ) {
     fun toCoseSignatureInput(
         externalAad: ByteArray = byteArrayOf(),
+        detachedPayload: ByteArray? = null,
     ): ByteArray = CoseSignatureInput(
         contextString = "Signature1",
         protectedHeader = protectedHeader.toZeroLengthByteString(),
         externalAad = externalAad,
-        payload = payload,
+        payload = payload ?: detachedPayload,
```

Comment:
This seems off: we have a payload member, but can also use a detached payload, which means we can create a CoseSigned with a payload and then pass a detached payload in this function, which will then be ignored. That is a footgun.

Reply:
Changed that

Reply:
How about this: Instead of passing a detached payload, just make the last parameter a flag:
```kotlin
fun toCoseSignatureInputWithDetachedPayload(
        externalAad: ByteArray = byteArrayOf(),
        detachPayload: Boolean = false,
    ): ByteArray = CoseSignatureInput(
        contextString = "Signature1",
        protectedHeader = protectedHeader.toZeroLengthByteString(),
        externalAad = externalAad,
        payload = if(detachPayload) null else payload,
        ).serialize()
```

No more ambiguities this way.

Reply:
My comment was, of course misleading. The new function should go and the old one should have the parameter:

```kotlin
fun toCoseSignatureInput(
    externalAad: ByteArray = byteArrayOf(),
    detachPayload: Boolean = false,
): ByteArray = CoseSignatureInput(
        contextString = "Signature1",
        protectedHeader = protectedHeader.toZeroLengthByteString(),
        externalAad = externalAad,
        payload = if(detachPayload) null else payload,
        ).serialize()
```

Reply:
Refactored as discussed offline